physicsnemo/experimental/models/dit/dit.py

```diff
@@ -186,24 +183,47 @@ def __init__(
             self.patch_size[0] * self.patch_size[1] * self.out_channels,
             layernorm_backbone,
         )
-        self.patch_recovery = PatchRecovery2D(
-            self.input_size,
-            self.patch_size,
-            self.patch_size[0] * self.patch_size[1] * self.out_channels,
-            self.out_channels,
-        )
+        self.initialize_weights()
+
+
+    def initialize_weights(self):
+        # Apply a basic Xavier uniform initialization to all linear layers.
+        def _basic_init(module):
+            if isinstance(module, nn.Linear):
+                torch.nn.init.xavier_uniform_(module.weight)
+                if module.bias is not None:
+                    nn.init.constant_(module.bias, 0)
+        self.apply(_basic_init)
+
+        # Initialize the learnable positional embedding with a normal distribution.
+        nn.init.normal_(self.pos_embed, std=0.02)
+
+        # Initialize the patch embedding projection (a Conv2D layer).
+        w = self.x_embedder.proj.weight.data
+        nn.init.xavier_uniform_(w.view([w.shape[0], -1]))
+        if self.x_embedder.proj.bias is not None:
+            nn.init.constant_(self.x_embedder.proj.bias, 0)
+        
+        # DiT specific initializations
+        for block in self.blocks:
+            nn.init.constant_(block.adaptive_modulation[-1].weight, 0)
+            nn.init.constant_(block.adaptive_modulation[-1].bias, 0)
+
+        nn.init.constant_(self.proj_layer.adaptive_modulation[-1].weight, 0)
+        nn.init.constant_(self.proj_layer.adaptive_modulation[-1].bias, 0)
+        
+        nn.init.constant_(self.proj_layer.output_projection.weight, 0)
+        nn.init.constant_(self.proj_layer.output_projection.bias, 0)
```

Comment:
If these dedicated inits are not actually needed, I would suggest removing them for cleanliness/clarity. The pos embed init can be done in `__init__` by `nn.Parameter(torch.randn(...))`